### ISSUE TITLE
`require_trailing_commas` is not triggered on switch expressions

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/linter#4982]
**Describe the issue**
`require_trailing_commas` is not triggered on switch expressions.

**To Reproduce**

```dart
enum E {
  a,
  b,
  c,
  d,
  e,
  f;
}

int a(E e) {
  return switch (e) {
    E.a => 0,
    E.b => 1,
    E.c => 2,
    E.d => 3,
    E.e => 4,
    E.f => 5     // Should trigger
  };
}
```

**Expected behavior**
Triggering of `require_trailing_commas`.

**Comment by FMorschel** [comment]
Also, the current assist to transform switch statements into switch expressions doesn't place a trailing comma there even if the lint is enabled.

If the reviewer agrees, I can make a new issue if necessary.

**Comment by bwilkerson** [comment]
I'm not sure we want to trigger the lint in this case.

My understanding is that the lint exists because of the way a trailing comma in a parameter or argument list impacts the formatter, and some users wanted their code to always be formatted per the trailing-comma style. To the best of my knowledge, the formatter behaves the same whether of not there's a trailing comma in a switch expression, making this an odd choice.

If the formatter is updated so that the trailing comma no longer impacts the behavior, then we're likely to consider deprecating this lint.

**Comment by FMorschel** [comment]
I'm sorry but it behaves differently if the `switch expression` is short:

```dart
enum E {
  a,
  b,
  c;
}

int a(E e) {
  return switch (e) { E.a => 0, E.b => 1, E.c => 2 };
}
```

I'm sorry if I was not clear enough to start.

**Comment by bwilkerson** [comment]
Thanks, I don't think I've written a switch expression short enough to fit on one line.

My expectation, given that, is that the future release of the formatter will likewise add and remove the trailing comma from switch expressions, making it unnecessary to extend this lint rule to support that case.
